apps/api/src/app/workflows-v2/usecases/preview/services/payload-merger.service.ts

```diff
@@ -133,41 +155,55 @@ export class PayloadMergerService {
 
     mergedPayload.subscriber = _.merge({}, fullSubscriberSchema, userSubscriberData);
 
-    /*
-     * Preserve steps from payloadExample (which contains correctly generated digest events)
-     * and merge with user-provided steps and mock data for missing steps
-     */
-    const stepsFromPayloadExample = (payloadExample.steps as Record<string, unknown>) || {};
-    const generatedStepsObject = await this.createFullStepsObject(workflow, command, userPayloadExample);
-
-    /*
-     * Merge with priority: user steps > payloadExample steps > generated mock steps
-     * Use mergeWith to ensure user-provided data (including empty objects) takes precedence
-     */
-    mergedPayload.steps = _.mergeWith(
-      {},
-      generatedStepsObject,
-      stepsFromPayloadExample,
-      (userPayloadExample?.steps as Record<string, unknown>) || {},
-      (objValue, srcValue) => {
-        // If source value is provided by user, always use it (even if it's an empty object)
-        if (srcValue !== undefined) {
-          return srcValue;
-        }
+    if (workflow && stepIdOrInternalId) {
```

Comment:
only when the workflow and step details are passed create the `steps` in the payload example